core/src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -702,49 +709,83 @@ public void finalizeTriggerables() throws IllegalStateException {
      * Get all of the elements which will need to be evaluated (in order) when the
      * triggerable is fired.
      */
-    private void fillTriggeredElements(Triggerable t, Vector<Triggerable> destination) {
+    private void fillTriggeredElements(Triggerable t,
+                                       Vector<Triggerable> destination,
+                                       boolean isRepeatEntryInit) {
         if (t.canCascade()) {
-            for (int j = 0; j < t.getTargets().size(); j++) {
-                TreeReference target = (TreeReference)t.getTargets().elementAt(j);
-                Vector<TreeReference> updatedNodes = new Vector<TreeReference>();
-                updatedNodes.addElement(target);
-
-                //For certain types of triggerables, the update will affect not only the target, but
-                //also the children of the target. In that case, we want to add all of those nodes
-                //to the list of updated elements as well. For instances, relevancy of a parent will
-                // require triggers pointing to children to need to be recalcualted
-                if (t.isCascadingToChildren()) {
-                    addChildrenOfReference(target, updatedNodes);
+            for (TreeReference target : t.getTargets()) {
+                Vector<TreeReference> updatedNodes;
+                if (isRepeatEntryInit) {
+                    // children have already been added to 'destination'
```

Comment:
If the above, bear in mind that there could be references to the triggerable conditions _outside_ of the repeat, which wouldn't be triggered based on this.,